src/renderer/components/commands-editors.tsx

```diff
@@ -143,24 +140,22 @@ export class EditorDropdown extends React.Component<
       const { editorMosaic } = appState;
       editorMosaic.addNewFile(id);
       editorMosaic.show(id);
-      editorMosaic.customMosaics.push(id);
     } catch (error) {
       appState.showErrorDialog(error.message);
     }
-  }
+  };
 
-  public async removeCustomEditor(event: React.MouseEvent) {
+  public removeFile = (event: React.MouseEvent) => {
     const { id } = event.currentTarget;
     const { editorMosaic } = this.props.appState;
 
-    console.log(`EditorDropdown: Removing custom editor ${id}`);
-    editorMosaic.removeCustomMosaic(id as EditorId);
-  }
+    editorMosaic.hide(id as EditorId);
```

Comment:
thought: the previous implementation of `removeCustomMosaic` also entirely removed the editor from the `customEditors` mosaic. Here, we're instead only hiding it. Is there a further action we can take to ensure that the editor is entirely erased?

Reply:
Previous implementation:

Reply:
Previous implementation of removeCustomMosaic():

```ts
  @action public removeCustomMosaic(id: EditorId) {
    this.hide(id);
    this.customMosaics = this.customMosaics.filter((mosaic) => mosaic !== id);
  }
```

But the thing is, `EditorMosaic.customMosaics` [isn't being used anywhere](https://github.com/electron/fiddle/search?q=customMosaics), so the function is equivalent to calling `hide()`. No further action is needed.

Slightly more info: EditorMosaic.hide() makes a backup of the monaco editor's state and then removes it from the mosaic property. That mosaic property is an observable and _that_ is what triggers react-mosaic-component to actually initiate creating / deleting monaco editors. EditorMosaic doesn't actually create or destroy the directly; it's all done indirectly via the mosaic component. This is the same approach that Fiddle's been using forever.

Reply:
I see. This seems to be a wider problem with the original "remove" functionality because removing a custom editor just hides it but it remains in the `Editors` dropdown regardless.

I filed a separate issue describing my qualms with this UX: https://github.com/electron/fiddle/issues/863

Reply:
@ckerr I think removing the Popover `Remove` menu altogether might be worth it because:

1) If the editor is visible, `editor.hide()` is the same as `editor.toggle()` and we can already toggle by clicking on the editor name directly.
2) If the editor is not visible, `editor.hide()` is a no-op.

Reply:
This is a good point. We need a way to actually remove a editor _and_ its backed up state if the user doesn't want that file anymore, and I don't think we have that right now.